flashlight/fl/tensor/backend/jit/JitTensorBase.cpp

```diff
@@ -23,6 +23,22 @@ JitTensorBase::~JitTensorBase() {
   node_->decRefCount();
 }
 
+// take care of refcount for old & new nodes
+void JitTensorBase::replaceNode(Node* newNode) const {
+  if (node_ != newNode) {
+    node_->decRefCount();
+    newNode->incRefCount();
+    const_cast<JitTensorBase*>(this)->node_ = newNode;
```

Comment:
Welp. I think the root cause of this is that `TensorBackend::eval(const Tensor& tensor)`. But upon materialization I want to use optimizer to update JitTensor's `node` here.

I think the `const` in `TensorBackend::eval` really applies to the "Tensor evaluation result" (which is indeed unchanged here). But C++ `const` doesn't give us enough granularity to specify that...

Reply:
@StrongerXi that makes a lot of sense. Many methods in `TensorAdapter` are already intentionally non-const for this reason (because even inspection might modify internal state) -- feel free to send a separate change to make this non-const.

cc @bwasti for an example of something that we want `const` in the public API but don't want `const` internally